### ISSUE TITLE
ansible variables group_names and group_names from hostvars have different content when printed

### ISSUE THREAD

**Issue by archf** [ansible/ansible#17263]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

Facts gathering and inventory variables.
##### ANSIBLE VERSION

```
ansible --version
ansible 2.1.1.0
```
##### CONFIGURATION

Vanilla.
##### OS / ENVIRONMENT

Ubuntu 16.04 (managing localhost)
##### SUMMARY

Difference in group_names from `hostvars` vs `group_names`.
##### STEPS TO REPRODUCE

For this given inventory:

```
[foo]
localhost ansible_host=localhost ansible_connection=local
```

And this play:

```

---

- hosts: foo

  tasks:
    - debug: var=hostvars
    - debug: var=group_names
```

Invoking like such:

```
 ansible-playbook -i local play.yml
```
##### EXPECTED RESULTS

Content of `group_names` should be identical. In the end, ansible behavior is ok because it uses `- debug: var=group_names` content. But this is rather confusing when examining `hostvars` variable for debugging purpose. Did i miss something?
##### ACTUAL RESULTS

Long story short here's the value from the task `- debug: var=group_names`
TASK [debug] *******************************************************************
ok: [localhost] => {
    "group_names": [
        "foo"
    ]
}
and truncated from hostvars (see the full ouptut in this [gist](https://gist.github.com/archf/ecac59cef8dd7a6bd861001e70040bc6))

```
            "group_names": [
                "ungrouped"
            ], 
            "groups": {
                "all": [
                    "localhost"
                ], 
                "foo": [
                    "localhost"
                ], 
                "ungrouped": [
                    "localhost"
                ]
            }, 
```

**Comment by bcoca** [comment]
This was due to bug or localhost always being added again to inventory, should be fixed in devel and upcoming 2.1.2 (rc1 is already out).
